team-manager/src/main/java/ua/knu/knudev/teammanager/service/AccountProfileService.java

```diff
@@ -2,52 +2,103 @@
 
 import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import ua.knu.knudev.fileserviceapi.api.FileServiceApi;
 import ua.knu.knudev.knudevsecurityapi.api.AccountAuthServiceApi;
 import ua.knu.knudev.knudevsecurityapi.request.AccountCreationRequest;
-import ua.knu.knudev.knudevsecurityapi.response.AccountCreationResponse;
+import ua.knu.knudev.knudevsecurityapi.response.AuthAccountCreationResponse;
 import ua.knu.knudev.teammanager.domain.AccountProfile;
+import ua.knu.knudev.teammanager.domain.Department;
+import ua.knu.knudev.teammanager.domain.Specialty;
 import ua.knu.knudev.teammanager.mapper.AccountProfileMapper;
 import ua.knu.knudev.teammanager.repository.AccountProfileRepository;
 import ua.knu.knudev.teammanagerapi.api.AccountProfileApi;
+import ua.knu.knudev.teammanagerapi.dto.AcademicUnitsIds;
 import ua.knu.knudev.teammanagerapi.dto.AccountProfileDto;
 import ua.knu.knudev.teammanagerapi.exception.AccountException;
 import ua.knu.knudev.teammanagerapi.response.AccountRegistrationResponse;
 
 @Service
 @RequiredArgsConstructor
 public class AccountProfileService implements AccountProfileApi {
+
     private final AccountProfileRepository accountProfileRepository;
     private final AccountProfileMapper accountProfileMapper;
     private final AccountAuthServiceApi accountAuthServiceApi;
+    private final FileServiceApi fileServiceApi;
+    private final DepartmentService departmentService;
 
     @Override
     public AccountProfileDto getByEmail(String email) {
+        // Implementation not provided
         return null;
     }
 
     @Override
-    public AccountRegistrationResponse createAccount(AccountCreationRequest accountCreationRequest) {
-        if (accountExists(accountCreationRequest.email())) {
-            throw new AccountException("Account already exists");
-        }
+    @Transactional
```

Comment:
In other way it can make more work than we need, so if it work without transactional without errors, delete must have

Reply:
So, it might work without it, and in fact it will, but think of the quantity of transactions. If making without @Transactional, then the quantity will be ~5. Its not, what we wont. And we dont want to create the avatar of account without account itself being created ( see A in ACID principles of relational databases ). A - atomicity. All or nothing